packages/base/src/api/client.ts

```diff
@@ -38,7 +59,8 @@ export abstract class BaseApiClient {
       // this means ID token has expired so we'll recreate session and try again
       if (error.response && error.response.status === 401 && error.response.statusText === 'Unauthorized') {
         this.api = undefined;
-        const api = await this.init();
+
+        const api = await this.refresh();
```

Comment:
This is a bit of a nitpick, but it seems weird to me that in the first invocation of `apiCall`, `this.refresh` would be called. It seems like `this.init` would be called and it would only attempt to refresh in the case that `this.session` already exists

Reply:
This logic would be executed in error is returned with 401 status code and in case it contains specific statusText.
So it should be called only when access token expires. We fallback to `init` mechanism in cases when refresh mechanism is not successful.